datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -332,34 +386,46 @@ fn group_aggregate_batch(
 
                     group_state
                         .indices
-                        .push_accounted(row as u32, &mut allocated); // remember this row
+                        .push_accounted(row as u32, &mut row_allocated); // remember this row
                 }
                 //  1.2 Need to create new entry
                 None => {
+                    let accumulator_set =
+                        aggregates::create_accumulators(normal_aggr_expr)?;
                     // Add new entry to group_states and save newly created index
                     let group_state = RowGroupState {
                         group_by_values: group_rows.row(row).owned(),
                         aggregation_buffer: vec![0; state_layout.fixed_part_width()],
+                        accumulator_set,
                         indices: vec![row as u32], // 1.3
                     };
-                    let group_idx = group_states.len();
+                    let group_idx = row_group_states.len();
 
                     // NOTE: do NOT include the `RowGroupState` struct size in here because this is captured by
                     // `group_states` (see allocation down below)
-                    allocated += (std::mem::size_of::<u8>()
+                    row_allocated += (std::mem::size_of::<u8>()
                         * group_state.group_by_values.as_ref().len())
                         + (std::mem::size_of::<u8>()
                             * group_state.aggregation_buffer.capacity())
                         + (std::mem::size_of::<u32>() * group_state.indices.capacity());
 
+                    // Allocation done by normal accumulators
+                    normal_allocated += (std::mem::size_of::<Box<dyn Accumulator>>()
```

Comment:
Why is there a 2nd memory tracking variable? Also, as far as I can tell, this is never used (you only add to it but never read it).

Reply:
Removed redundant variable.